pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/InMemoryDelayedDeliveryTracker.java

```diff
@@ -157,18 +224,27 @@ public CompletableFuture<Void> clear() {
 
     @Override
     public long getNumberOfDelayedMessages() {
-        return priorityQueue.size();
+        return priorityQueue.values().stream().mapToLong(
+                ledgerMap -> ledgerMap.values().stream().mapToLong(
+                        Roaring64Bitmap::getLongCardinality).sum()).sum();
     }
 
+    /**
+     * This method rely on Roaring64Bitmap::getLongSizeInBytes to calculate the memory usage of the buffer.
+     * The memory usage of the buffer is not accurate, because Roaring64Bitmap::getLongSizeInBytes will
+     * overestimate the memory usage of the buffer a lot.
+     * @return the memory usage of the buffer
+     */
     @Override
     public long getBufferMemoryUsage() {
-        return priorityQueue.bytesCapacity();
+        return priorityQueue.values().stream().mapToLong(
+                ledgerMap -> ledgerMap.values().stream().mapToLong(
+                        Roaring64Bitmap::getLongSizeInBytes).sum()).sum();
     }
```

Comment:
Does this method get called in the same thread as other InMemoryDelayedDeliveryTracker methods? If not, there's a need to make the methods synchronized or add a ReadWriteLock to fix thread safety.

Reply:
`TripleLongPriorityQueue` is not thread safe too.
`getBufferMemoryUsage` is used to print the stat used by `InMemoryDelayedDeliveryTracker`, which don't require too much accuracy. 
What is more, the method `Roaring64Bitmap::getLongSizeInBytes` is not accurate too, as stated in the comment.